src/status_im/contexts/communities/actions/accounts_selection/view.cljs

```diff
@@ -86,7 +89,9 @@
         :track-text          (i18n/label :t/slide-to-request-to-join)
         :track-icon          :i/face-id
         :customization-color color
-        :on-complete         #(join-community-and-navigate-back id selected-permission-addresses)}]]]))
+        :on-complete         #(join-community-and-navigate-back id
+                                                                selected-permission-addresses
```

Comment:
We don't need to pass values that are already stored in the app-db to the event `:communities/request-to-join`. For example, the value of `airdrop-address` is computed solely from the subscription `:communities/airdrop-address`, which means the event can easily grab this value from the app-db. The same idea applies to `selected-permission-addresses`.

Reply:
maybe is ok in this case, but in general I think on-press event should try to use  parameters (or closures in this case) that are passed in the render, rather than relying on the state of the database, to the extent of where that's useful and feasible and does not occur in extra subscriptions for example.

I think it's a better to make events more independent of the database, that makes it more explicit and clearer, as otherwise there's implicit parameters being passed and it's bit harder to inspect, also there's always the possibility that the value could change after the user confirms (not in this case I guess).
 
In this case might be totally fine, since as far as I can see, we only subscribe to that piece of data to pass it as a parameter, but as a general rule, I think it's clearer if we tend to be more declarative and less dependent on the current state of the database in events (We have similar issues with `current-chat-id`, which is always an implicit parameter and we struggle to keep it synchronized with the events, although of course it's a much more complex scenario).

Reply:
I have updated the events to use values from the database, but I think it's better to write new events to keep the old and new flow separate. Do you have any resource suggestions regarding best practices on this topic (events, subscriptions, etc.)?

Reply:
@ajayesivan, in terms of best practices, I think the re-frame docs generally explain and document why certain choices should be made, but there's more than one truth, so it's up to each developer to judge each context. Based on my experience, good Redux code also translates well to re-frame, so if you have experience with it, perhaps there are good Redux sources out there.

@cammellos, I don't know the specifics of the issues with `current-chat-id`, but I can see why we may have/had bugs related to this. Probably places that should've received an explicit ID, instead looked for the current chat ID (?). But this is a problem we can't have here, because there's only one flow the user can select addresses to join and no other screen should refer to this app-db state for anything else.

> also there's always the possibility that the value could change after the user confirms (not in this case I guess).

That's why I think we should pretty much always strive to only pass stable identifiers to subscriptions and events (like IDs, or addresses). Anything that can be derived from them should not be passed because then we can have this mid-flight state change problem you mentioned, among other problems.

And you're correct, for Ajay's PR it's fine because we are passing addresses.

> as otherwise there's implicit parameters being passed and it's bit harder to inspect

👍🏼 Yes, sometimes this is a pain. Other times, when events receive fewer arguments, it makes debugging them with the REPL easier because I can check what they do without having to use re-frisk or other techniques to print the arguments being passed from views.